src/anomalib/config/config.py

```diff
@@ -7,61 +7,135 @@
     """Update config with image size as tuple, effective input size and tiling stride.
 
     Convert integer image size parameters into tuples, calculate the effective input size based on image size
     and crop size, and set tiling stride if undefined.
 
     Args:
-        config (DictConfig | ListConfig): Configurable parameters object
+        config (DictConfig | ListConfig | Namespace): Configurable parameters object
 
     Returns:
         DictConfig | ListConfig: Configurable parameters with updated values
     """
     # Image size: Ensure value is in the form [height, width]
-    image_size = config.dataset.get("image_size")
+    image_size = config.data.init_args.get("image_size")
     if isinstance(image_size, int):
-        config.dataset.image_size = (image_size,) * 2
-    elif isinstance(image_size, ListConfig):
+        config.data.init_args.image_size = (image_size,) * 2
+    elif isinstance(image_size, (ListConfig, tuple)):
         assert len(image_size) == 2, "image_size must be a single integer or tuple of length 2 for width and height."
     else:
         raise ValueError(f"image_size must be either int or ListConfig, got {type(image_size)}")
 
-    # Center crop: Ensure value is in the form [height, width], and update input_size
-    center_crop = config.dataset.get("center_crop")
-    if center_crop is None:
-        config.model.input_size = config.dataset.image_size
-    elif isinstance(center_crop, int):
-        config.dataset.center_crop = (center_crop,) * 2
-        config.model.input_size = config.dataset.center_crop
-    elif isinstance(center_crop, ListConfig):
-        assert len(center_crop) == 2, "center_crop must be a single integer or tuple of length 2 for width and height."
-        config.model.input_size = center_crop
-    else:
-        raise ValueError(f"center_crop must be either int or ListConfig, got {type(center_crop)}")
+    # Use input size from data to model input. If model input size is defined, warn and override.
+    # If input_size is not part of the model parameters, remove it from the config. This is required due to argument
+    # linking from the cli.
+    model_module = import_module(".".join(config.model.class_path.split(".")[:-1]))
+    model_class = getattr(model_module, config.model.class_path.split(".")[-1])
+
+    # Assign center crop
+    center_crop = config.data.init_args.get("center_crop", None)
+    if center_crop:
+        config.data.init_args.center_crop = to_tuple(center_crop)
+    config.data.init_args.image_size = to_tuple(config.data.init_args.image_size)
+
+    if "input_size" in inspect.signature(model_class).parameters:
+        if "input_size" in config.model.init_args:
+            # Center crop: Ensure value is in the form [height, width], and update input_size
+            if center_crop is not None:
+                config.model.init_args.input_size = center_crop
+                logger.info(f"Setting model size to crop size {center_crop}")
```

Comment:
I think some models build layers based on the input size so not sure if transforms will help here. Also, we might have to modify this in the future. This might not work well with Tiling.